packages/ui-components/src/components/Manage/DomainProfileModal.tsx

```diff
@@ -67,6 +89,7 @@ export type DomainProfileModalProps = {
   address?: string;
   domain: string;
   open: boolean;
+  metadata?: Record<string, string | boolean>;
```

Comment:
You may benefit from some more refined types in the long run:
```ts
export enum Blockchain {
  ETH = 'ETH',
  MATIC = 'MATIC',
}

export enum Registry {
  ZNS = 'ZNS',
  CNS = 'CNS',
  UNS = 'UNS',
  ENS = 'ENS',
  DNS = 'DNS',
}

export type Meta = {
  blockchain: Blockchain | null;
  domain: string;
  logicalOwnerAddress: string | null;
  namehash: string;
  networkId: number | null;
  owner: string | null;
  registryAddress: string | null;
  resolver: string | null;
  reverse: boolean;
  tokenId?: string;
  ttl: number;
  type: Registry;
};
```

```suggestion
  metadata: Meta;
```

Reply:
Yeah, I may need to move to stronger type if I keep referencing the metadata. Used to just be a proxy from resolution service, so wasn't important to know the contents.